components/brave_wallet_ui/common/async/brave_rewards_api_proxy.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import {
+  UserType,
+  userTypeFromString
+} from '../../../brave_rewards/resources/shared/lib/user_type'
+
+import {
+  ExternalWallet,
+  externalWalletFromExtensionData,
+} from '../../../brave_rewards/resources/shared/lib/external_wallet'
+
+
+export const WalletStatus = {
+  kNotConnected: 0,
+  kConnected: 2,
+  kLoggedOut: 4,
+} as const
+export type WalletStatus = typeof WalletStatus[keyof typeof WalletStatus]
+
+export type RewardsWallet = Pick<
```

Comment:
Note that within the rewards codebase `RewardsWallet` means something a different from `ExternalWallet` (for historical reasons). I would choose a different name for your custom interface here. `RewardsExternalWallet` (for example) would be fine.